awsplugins/ec2/ec2.go

```diff
@@ -9,35 +9,115 @@
 package ec2
 
 import (
-	"github.com/aws/aws-sdk-go/aws/ec2metadata"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"bytes"
+	"encoding/json"
+	"net/http"
+
 	"github.com/aws/aws-xray-sdk-go/internal/logger"
 	"github.com/aws/aws-xray-sdk-go/internal/plugins"
 )
 
 // Origin is the type of AWS resource that runs your application.
 const Origin = "AWS::EC2::Instance"
 
-// Init activates EC2Plugin at runtime.
+type Metdata struct {
```

Comment:
For performance reasons, we probably shouldn't record the attributes we don't use. We should only need the 4 of these we record.

Reply:
Yes. Makes sense!